src/Integration/LocalServices/SolutionBindingInformationProvider.cs

```diff
@@ -31,13 +31,46 @@ public SolutionBindingInformationProvider(IServiceProvider serviceProvider)
         }
 
         #region ISolutionBindingInformationProvider
+        public bool IsSolutionBound()
+        {
+            return this.GetSolutionBinding() != null;
+        }
+
         public IEnumerable<Project> GetBoundProjects()
         {
+            return this.GetBoundProjects(this.GetSolutionBinding());
+        }
+
+        public IEnumerable<Project> GetUnboundProjects()
+        {
+            return this.GetUnBoundProject(this.GetSolutionBinding());
```

Comment:
Why the extra layer of indirection here? You're not actually exposing the `this.GetUnBoundProject(BoundSonarQubeProject)` method for testing with a passed in BoundSonarQubeProject.